backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java

```diff
@@ -233,6 +244,31 @@
           .append(" AND oe.field_type = :field_type")
           .toString();
 
+  private LoadingCache<String, ReadWriteLock> locks =
+      CacheBuilder.newBuilder()
+          .maximumSize(CACHE_SIZE)
+          .expireAfterWrite(DURATION, TimeUnit.MINUTES)
+          .build(
+              new CacheLoader<String, ReadWriteLock>() {
+                public ReadWriteLock load(String lockKey) {
+                  return new ReentrantReadWriteLock() {};
+                }
+              });
+
+  protected AutoCloseable acquireReadLock(String lockKey) throws ExecutionException {
+    ReadWriteLock lock = locks.get(lockKey);
```

Comment:
do you think a log statement here and the function below will help us in the future to debug a particular slow call which probably was waiting on a lock?

Reply:
Sure I can add it.